src/main/java/io/cdap/plugin/kudu/KuduSink.java

```
@@ -222,27 +215,23 @@ private void createKuduTable(FailureCollector failureCollector) {
             localClient.createTable(kuduSinkConfig.getTableName(), kuduSchema, options);
           LOG.info("Successfully create Kudu table '%s', Table ID '%s'", kuduSinkConfig.getTableName(), table.getTableId());
         } catch (KuduException e) {
-          failureCollector.addFailure(
-            String.format("Unable to create table '%s'.", kuduSinkConfig.getTableName()),null)
-            .withConfigProperty(KuduSinkConfig.NAME)
-            .withStacktrace(e.getStackTrace());
+          throw new RuntimeException(
+            String.format("Unable to create table '%s'. Reason : %s", kuduSinkConfig.getTableName(), e.getMessage())
+          );
         }
       } else {
         // If the table exists in Kudu, compare the schema and make sure they are the same.
         // If they are not the same then throw an exception.
         KuduTable table = localClient.openTable(kuduSinkConfig.getTableName());
         org.apache.kudu.Schema kuduSchema = table.getSchema();
-        checkSchemaCompatibility(kuduSchema, writeSchema, failureCollector);
+        checkSchemaCompatibility(kuduSchema, writeSchema);
       }
     } catch (KuduException e) {
-      failureCollector.addFailure(
-        String.format("Unable to check if the table '%s' exists in Kudu.", kuduSinkConfig.getTableName()),null)
-        .withConfigProperty(KuduSinkConfig.NAME)
-        .withStacktrace(e.getStackTrace());
+      String msg = String.format("Unable to check if the table '%s' exists in kudu. Reason : %s",
+                                 kuduSinkConfig.getTableName(), e.getMessage());
+      LOG.warn(msg);
```

catch, log, and rethrow is a bad pattern. It usually results in multiple log messages about the same thing when the caller logs the exception that they catch. You can just throw the exception without logging.

---

Done.